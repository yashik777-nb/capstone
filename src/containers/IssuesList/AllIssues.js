import React from "react";
import { connect } from "react-redux";
import { Container } from "@material-ui/core";
import { withStyles } from "@material-ui/styles";
import Grid from "@material-ui/core/Grid";
import CustomizeFields from "./CustomizeFields";
import CardIssue from "../../components/Issue/CardIssue";
import IssuesChart from "../../containers/Charts/IssuesChart";

const styles = (theme) => ({
  root: {
    flexGrow: 1,
  },
});
class AllIssues extends React.Component {
  render() {
    const { classes } = this.props;
    const issuesLength = this.props.issues.length;
    let issueCards = this.props.issues.map((issue) => {
      return (
        <CardIssue
          key={issue.id}
          {...issue}
          authenticated={this.props.authenticated}
          customize={this.props.customize}
        />
      );
    });

    return (
      <div className="m-4">
        <Container maxWidth="lg">
          {this.props.authenticated ? (
            <h4
              style={{ marginTop: 20, textAlign: "center", color: "#3f51b5" }}
            >
              Hello {this.props.firstname}! Please find the below issues
            </h4>
          ) : null}

          {issuesLength > 0 ? (
            <div>
              <CustomizeFields />
              <div className={classes.root}>
                <Grid>{issueCards}</Grid>
              </div>
              <IssuesChart />
            </div>
          ) : (
            <h5 style={{ marginTop: 20, textAlign: "center", color: "red" }}>
              No Data Found
            </h5>
          )}
        </Container>
      </div>
    );
  }
}

// mapStateToProps
const mapStateToProps = (state) => {
  return {
    issues: state.issues.issues,
    authenticated: state.users.authenticated,
    firstname: state.users.firstname,
    customize: state.customize,
  };
};

export default connect(mapStateToProps)(
  withStyles(styles, { withTheme: true })(AllIssues)
);
