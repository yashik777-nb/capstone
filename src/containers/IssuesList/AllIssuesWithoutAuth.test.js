import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import AllIssues from "./AllIssues";

configure({ adapter: new Adapter() });

const styles = {
  root: "",
};

const store = {
  issues: {
    issues: [
      {
        id: 1,
        title: "Issue One",
        issueDescription: "Issue Description One",
        severity: "Critical",
        status: "Open",
        createdDate: "28/02/2021  ",
        resolvedDate: "29/03/2021",
        views: 5,
        backGroundColor: "#7828BF",
        hoverBackGroundColor: "#9C5BB7",
      },
    ],
  },
  customize: {
    severity: true,
    status: true,
    createdDate: true,
    resolvedDate: true,
  },
  users: {
    authenticated: false,
    firstname: "TestFirstName",
  },
};

const props = {
  classes: styles,
};

const mockStore = configureStore();

describe("Navigation Bar Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <Provider store={mockStore(store)}>
        <BrowserRouter>
          <AllIssues {...props} />
        </BrowserRouter>
      </Provider>
    );
  });

  it("Should not render h4 tag if not authenticated", () => {
    expect(wrapper.find("h4").length).toEqual(0);
  });

  test("Should Render All Issues Without Auth", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
