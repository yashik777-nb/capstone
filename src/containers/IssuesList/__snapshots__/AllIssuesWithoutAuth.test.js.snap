// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar Tests Should Render All Issues Without Auth 1`] = `ReactWrapper {}`;
